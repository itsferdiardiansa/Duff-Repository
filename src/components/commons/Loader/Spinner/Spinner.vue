<template>
  <div :class="`${prefixClass}-spinner`" :style="getStyle">
    <div></div>
    <div></div>
    <div></div>
  </div>
</template>
<script>
import { computed } from 'vue';
export default {
  name: 'CMSpinner',
  props: {
    width: {
      type: [Number, String],
      default: 10,
    },
    height: {
      type: [Number, String],
      default: 10,
    },
  },
  setup(props) {
    const getStyle = computed(() => {
      const { width, height } = props;
      let style = {};

      style = {
        width: typeof width === 'number' ? `${width}px` : width,
        height: typeof height === 'number' ? `${height}px` : height,
      };

      return style;
    });

    return {
      getStyle,
    };
  },
};
</script>
<style lang="scss" scoped>
.#{$prefixClass}-spinner {
  display: inline-block;
  position: relative;
  height: 70%;

  div {
    box-sizing: border-box;
    display: block;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    border: 3px solid #fff;
    border-radius: 50%;
    animation: #{$prefixClass}-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
    border-color: #fff transparent transparent transparent;

    &::nth-child(1) {
      animation-delay: -0.45s;
    }

    &::nth-child(2) {
      animation-delay: -0.3s;
    }

    &::nth-child(3) {
      animation-delay: -0.15s;
    }
  }
}
@keyframes #{$prefixClass}-ring {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
</style>
