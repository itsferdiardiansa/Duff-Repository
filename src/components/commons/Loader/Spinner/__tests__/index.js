import { mount } from '@vue/test-utils';
import Spinner from '@common/Loader/Spinner';

describe('common/Loader/Spinner', () => {
  it('will render and match snapshot', () => {
    const props = {
      width: 10,
      height: 10,
    };

    const Component = mount(<Spinner {...props} />);

    expect(Component.element).toMatchSnapshot();
  });
});
