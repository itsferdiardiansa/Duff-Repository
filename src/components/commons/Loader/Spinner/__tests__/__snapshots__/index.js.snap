// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Loader/Spinner will render and match snapshot 1`] = `
<div
  class="ss-spinner"
  style="width: 10px; height: 10px;"
>
  <div />
  <div />
  <div />
</div>
`;
